base/src/main/java/com/example/android/whileinuselocation/MainActivity.kt

```diff
@@ -35,34 +35,38 @@ import android.widget.TextView
 
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.app.ActivityCompat
-import androidx.core.os.BuildCompat
 import androidx.localbroadcastmanager.content.LocalBroadcastManager
 
 import com.google.android.material.snackbar.Snackbar
+
+private const val TAG = "MainActivity"
+private const val REQUEST_FOREGROUND_ONLY_PERMISSIONS_REQUEST_CODE = 34
+
 /**
- *  This app allows a user to track their location.
+ *  This app allows a user to receive location updates without the background permission even when
+ *  the app isn't in focus. This is the preferred approach for Android.
  *
- *  One or the features in the app creates a foreground service (tied to a Notification) when the
+ *  It does this by creating a foreground service (tied to a Notification) when the
  *  user navigates away from the app. Because of this, it only needs foreground or "while in use"
  *  location permissions. That is, there is no need to ask for location in the background (which
  *  requires additional permissions in the manifest).
  *
- *  There is another feature that does require location in both the foreground and background to
- *  show the proper way to get location for that scenario.
+ *  Note: Users have four options in Android 11+ regarding location:
  *
- *  Note: Users have three options in Android 10+ regarding location:
- *
- *  * Allow all the time
- *  * Allow while app is in use, i.e., while app is in foreground
+ *  * Allow while app is in use, i.e., while app is in foreground (new in Android 10)
+ *  * Allow one time use (new in Android 11)
  *  * Not allow location at all
+ *  * Allow all the time
```

Comment:
Move to the beginning of the list so that it's ordered from most to least permissive